code/sonalyze/application/local.go

```diff
@@ -35,7 +36,11 @@ func LocalOperation(command SampleAnalysisCommand, _ io.Reader, stdout, stderr i
 
 	var theLog db.SampleCluster
 	if len(args.LogFiles) > 0 {
-		theLog, err = db.OpenTransientSampleCluster(args.LogFiles, cfg)
+		ty := db.FileSampleCSV
```

Comment:
This snippet is used everywhere.  I think it should be factored, it should take two file type arguments and a list of files, and it should return either type or an indication that the files are not all of the same type.